yaksh/templates/exam.html

```diff
@@ -83,28 +78,56 @@
           <p class="text-center">Question(s) left: <b>{{ paper.questions_left }}</b></p>
           </div>
           <br>
-          <div class="">
-            <div class="">
-              <div class="">
-                {% for types in question_types %}
-                {% get_questions_by_type  paper.get_all_ordered_questions types as questions_by_type %}  
-                  {% if types == "mcq" %}  Single Correct Choice 
-                  {% elif types == "mcc" %}  Multiple Correct Choice 
-                  {% elif types == "code" %} Programming
-                  {% elif types == "upload" %} Assignment Upload 
-                  {% elif types == "integer" %} Integer Blanks
-                  {% elif types == "string" %} String Blanks
-                  {% elif types == "float" %} Float Blanks
-                  {% elif types == "arrange" %} Arranging Options  
-                  {% endif %}      
-                  --->
-                   <b>  {% for question in questions_by_type %} {{question.id}}  {%endfor%}  </b>
-                     <br>
-                {%endfor%}  
-              </div>
-            </div>
-          </div>
-        
+          <table class = "table table-bordered table-sm">
+            <thead>
+              <tr>
+                <th>Category</th>
+                <th>Question No.</th>
+              </tr>
+            </thead>
+            <tbody>
+              {% if index.objectives_index %}
+              <tr>
+                <td>
+                  Objectives:
+                </td>
+                <td>
+                  {{index.objectives_index|join:", "}}
+                </td>
+              </tr>
+              {% endif %}
+              {% if index.blanks_index %}
+              <tr>
+                <td>
+                  Blanks:
+                </td>
+                <td>
+                  {{index.blanks_index|join:", "}}
+                </td>
+              </tr>
+              {% endif %}
+              {% if index.upload_index %}
+              <tr>
+                <td>
+                  Upload:
+                </td>
+                <td>
+                  {{index.upload_index|join:", "}}
+                </td>
+              </tr>
+              {% endif %}
+              {% if index.programming_index %}
+              <tr>
+                <td>
+                  Programming:
+                </td>
+                <td>
+                  {{index.programming_index|join:", "}}
+                </td>
+              </tr>
+              {% endif %}
+            </tbody>
```

Comment:
I think it will be more than enough if we simply show objectives and programming!
Blanks are also objectives, and objective types will keep on adding in future, so better keep it simple for a student as well.
Upload type can be separate.